NSURL.m

```diff
@@ -351,7 +352,25 @@ static CFTypeRef CFURLCreatePropertyForKey(CFURLRef url, CFStringRef key, CFErro
     else if (CFEqual(key, kCFURLTypeIdentifierKey))
     {
         // Key for the resource’s uniform type identifier (UTI), returned as a CFString object.
+        void* LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
+        if(LaunchServicesHandle == NULL){
+            posixError(error);
+            return nil;
```

Comment:
should this actually return anything here though? This seems to be an outlier in the existing code and in my new code.

Reply:
The existing cases just all don't have that many steps. You do want to stop & not go any further if you're unable to `dlopen`. You could also jump out of the switch in some other way (some sort of `goto out;`), but a `return` also works, since that's what happens after the switch anyway.

So you also want to add `return`s to the failure branches below.